### DISCUSSION TITLE
How to hold left click.

### DISCUSSION THREAD

**Discussion by KlaasNiphaas** [PrismarineJS/mineflayer#2114] [Q&A] [answered]
Hey. I'm currently working on a bot that farms carrots on a skyblock island. I got it pretty much working with the moving and the timing of the moving but whenever i use bot.dig it either crashes when i have it dig every <250 ms. I have to turn it up for it to not crash after ~30 seconds. But when i turn it up it misses alot of crops. So holding left click would be much easier.

**Comment by rom1504** [comment, marked as answer]
holding left click is not something supported by minecraft protocol, it's emulated by minecraft clients
how do you dig repeatedly ? do you wait for the promise to be resolved ?

**Reply by KlaasNiphaas** [reply]
I have a while loop.

**Reply by KlaasNiphaas** [reply]
Also another question: is there a mineflayer Discord?

**Reply by rom1504** [reply]
a while loop with an await bot.dig inside ?

yes, the discord is linked in the readme

**Reply by KlaasNiphaas** [reply]
yea

**Reply by KlaasNiphaas** [reply]
Would await blockBreakProgressEnd(block) work?

**Reply by rom1504** [reply]
await bot.dig is enough, if it doesn't work there is a bug in your code
share your code if you want help

**Reply by KlaasNiphaas** [reply]
I'll try that. Ty for your help

**Reply by KlaasNiphaas** [reply]
Okay it worked. Again ty for your help

**Reply by zakarkaaaaa** [reply]
can you share your code please?
